bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.internal.module.handler;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.Action;
+import org.openhab.core.automation.handler.BaseActionModuleHandler;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.events.ItemEventFactory;
+import org.openhab.core.items.events.ItemStateEvent;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TypeParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an implementation of an ActionHandler. It send state updates for items.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class ItemUpdateActionHandler extends BaseActionModuleHandler {
+
+    public static final String ITEM_UPDATE_ACTION = "core.ItemUpdateAction";
+    public static final String ITEM_NAME = "itemName";
+    public static final String STATE = "state";
+
+    private final Logger logger = LoggerFactory.getLogger(ItemUpdateActionHandler.class);
+
+    private final EventPublisher eventPublisher;
+
+    private final Item item;
+    private final @Nullable State state;
+
+    /**
+     * constructs a new ItemCommandActionHandler
+     *
+     * @param module
+     * @param moduleTypes
+     */
+    public ItemUpdateActionHandler(Action module, EventPublisher eventPublisher, ItemRegistry itemRegistry) {
+        super(module);
+
+        this.eventPublisher = eventPublisher;
+
+        final Configuration config = module.getConfiguration();
+
+        final String itemName = (String) config.get(ITEM_NAME);
+        try {
+            this.item = itemRegistry.getItem(itemName);
+        } catch (ItemNotFoundException e) {
```

Comment:
The `ItemCommandActionHandler` does this check only within `execute`, but not within the constructor. I think that's the much better approach, because the item might not be defined at rule definition time, while this should not cause the rule to not initialize.